pkg/ddc/base/operation_lock.go

```diff
@@ -118,14 +96,7 @@ func ReleaseTargetDataset(ctx cruntime.ReconcileRequestContext, object client.Ob
 			// other error
 			return err
 		}
-		currentDataOpKey := dataset.GetDataOperationInProgress(operationTypeName)
-
-		if currentDataOpKey != GetDataOperationKey(object) {
-			ctx.Log.Info("Found Ref inconsistent with the reconciling DataBack, won't release this lock, ignore it", "Operation", operationTypeName, "ref", currentDataOpKey)
-			return nil
-		}
 		datasetToUpdate := dataset.DeepCopy()
-		datasetToUpdate.RemoveDataOperationInProgress(operationTypeName)
 
 		// different operation may set other fields
 		operation.RemoveTargetDatasetStatusInProgress(datasetToUpdate)
```

Comment:
If multiple data operations concurrently change targetDataset's status (e.g. DataMigrate resets dataset status to `Bound`), now it's possible that a DataMigrate is still running, but the Dataset has already been set back to `Bound`?